### PR TITLE
Use boto's high level upload_file() method

### LINKED ISSUE #116: EntityTooLarge error with 5+ GB upload
API docs say 10GB is allowed, but a user has reported hitting a limit in the SDK.

```python
File "/usr/local/lib/python2.7/dist-packages/mapbox/services/uploads.py", line 149, in upload
url = self.stage(fileobj)
File "/usr/local/lib/python2.7/dist-packages/mapbox/services/uploads.py", line 65, in stage
res = s3.Object(creds['bucket'], creds['key']).put(Body=fileobj)
File "/usr/local/lib/python2.7/dist-packages/boto3/resources/factory.py", line 498, in do_action
response = action(self, *args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/boto3/resources/action.py", line 83, in __call__
response = getattr(parent.meta.client, operation_name)(**params)
File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 228, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/usr/local/lib/python2.7/dist-packages/botocore/client.py", line 492, in _make_api_call
raise ClientError(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (EntityTooLarge) when calling the PutObject operation: Your proposed upload exceeds the maximum allowed size
```

### PR DESCRIPTION
**Pull request by sgillies**
Uses multipart uploading when helpful.

Closes #116

### CODE REVIEW

**Comment by sgillies** [comment]
@perrygeo `upload_file()` seems to be the ticket. How's this look to you?

The Distance API's results are changing and throwing off our tests, we'll need to address that in another ticket :(

File: mapbox/services/uploads.py [hunk 1]
```diff
@@ -62,7 +65,12 @@ def stage(self, fileobj, creds=None):
             region_name="us-east-1")
 
         s3 = session.resource('s3')
-        res = s3.Object(creds['bucket'], creds['key']).put(Body=fileobj)
+        # We'll use Bucket.upload_file() for actual files.
+        # Progress reporting for the CLI is a TODO.
+        if hasattr(fileobj, 'name') and os.path.exists(fileobj.name):
+            s3.Bucket(creds['bucket']).upload_file(fileobj.name, creds['key'])
```
Comment (perrygeo) [inline mapbox/services/uploads.py]: This feels a bit funny to require an open file handle only to access the object by its path. If it weren't for the pending changes in https://github.com/mapbox/mapbox-cli-py/pull/68 (stdin bytes accumulated client side then wrapped in a file-like object) I'd say just change the API to use a path. But I think this is a good solution for now.

**Comment by perrygeo** [comment]
mergin'. Hot fix for datasets API doctests coming shortly.

File: mapbox/services/uploads.py (hunk 1 above)
Comment (sgillies) [inline mapbox/services/uploads.py]: Totes.

Comment (perrygeo) [inline mapbox/services/uploads.py]: @sgillies what about using http://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Client.upload_fileobj here instead?

**Pull request merged**